packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -797,6 +1033,45 @@ void registerPlugins(Registrar registrar) {
 }
 ''';
 
+// TODO(egarciad): Evaluate merging the web and desktop plugin registry templates.
+const String _dartPluginRegistryForDesktopTemplate = '''
+//
+// Generated file. Do not edit.
+//
+
+import '{{mainEntrypoint}}' as entrypoint;
```

Comment:
What happens if the user adds a new plugin and does a hot reload/hot restart?
What happens if the user removes a plugin and does a hot restart?

In both cases the file needs to be regenerated, in the former its definitely a nice to have, but in a later that would require a quit/restart cycle to remove a plugin - because the compiler can't resolve it anymore.

What happens if the plugin declaration is invalid in a way that doesn't compile, what do those error messages look like?

Reply:
But in this case, if its a dart plugin and you remove it - you need to regenerate this file or else the app won't compile at all.

Reply:
so the case is:  as a user I have a Dart only plugin, run `flutter run`, then delete the dart plugin from pubspec.yaml, then hot restart?

So the workaround is to re-run `flutter run`.  I tried to regenerate the file in the recompile flow, but it required more refactoring.

Reply:
I think there could be more complicated scenarios too - like you update a version and that goes from being a Dart only plugin to not. Unfortunately the error message is going to be very difficult to understand, since the user is going to get a compilation error pointing them to code they didn't write

Reply:
yeah. I didn't know that `flutter run` supports adding/removing dependencies to pubpec.yaml if they don't need platform-side recompilation.  I also wonder if users are able to know the difference. e.g. If they consume a plugin, are they able to know that they can't do that for Android and iOS, but it might be ok for a Dart-only plugin.

Reply:
I think removing a plugin needs to be handled otherwise it will become a compilation error.

Reply:
Removing a plugin works, they would just need to type `flutter run` again.

Reply:
while not being able to add a new plugin is in-line with our current behavior on Android/iOS, losing the ability to remove a plugin without restarting the tool is definitely a regression.

Reply:
I started by handling this case (support adding and removing plugins). Unfortunately, it required additional changes to the test. See your comment about not introducing more globals or using `testUsingContext`.  My suggestion is two separate this feature into two PRs.